File: Group-03/API/src/main/java/com/example/demo/model/Order.java
package com.example.demo.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Data
@NoArgsConstructor       // adds default constructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    private String orderId;
    private Long userId;
    private Double totalPrice;
    private String orderStatus;
    private String paymentStatus;
    private String deliveryAddress;
    private String phoneNumber;
    private String paymentMethod;
    private String orderNotes;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime orderDate;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderProduct> items;

}

File: Group-03/UI/src/app/model/order.ts
export interface OrderProduct {
  productId: number;
  productTitle?: string;
  quantity: number;
  price: number;
}

export interface Order {
  id:number;
  orderId: string;
  userId: number;
  orderStatus: string;
  paymentStatus:string;
  paymentMethod:string;
  orderDate: string;
  totalPrice: number;
  deliveryAddress: string;
  phoNo: string;
  items: OrderProduct[];
  showDetails?: boolean;
}
